doc/utils/manifest.md

```diff
@@ -47,6 +51,52 @@ This section is optional. It contains metadata used to run application tests.
 | `unit_directory`   | Path of the directory where unit tests can be found                                            |
 | `pytest_directory` | Path of the directory where functional, Python test can be found (`conftest.py` file expected) |
 
+#### `[use_cases]`
+
+This section is optional. It contains metadata helping select build options depending on use cases
+The VSCode extension leverages this section to provide alternative build targets.
+
+| Field name   | Description                                                             |
+|--------------|-------------------------------------------------------------------------|
+| `<use_case>` | Options string : <ul><li>Environment variables definitions for C applications (e.g. `DEBUG=1`)</li><li>Valid Cargo build options for Rust applications (e.g. `--outdir mydir`)</li></ul> |
+
+This specify that in order to build for `<use_case>`, the options string must be provided in the build command line.
+You are free to add any use case you wish to have a VSCode build target for.
+The use case `default` refers to a standard build with no option. It is implicit and can't be redefined.
+
+Example:
+```
+[use_cases] # Coherent build options that make sense for your application
+debug = "DEBUG=1"
+test = "TESTING=1 TEST_PUBLIC_KEY=1"
+```
+
+#### `[test.dependencies.<test_use_case>]`
+
+The test.dependencies.* sections are optional. They contain metadata helping building side applications needed for your tests.
+You can define as many as you need.
+The VSCode extension leverages this field to build test dependencies for Speculos tests or device tests.
+
+| Field name | Description                                                                                       |
+|------------|---------------------------------------------------------------------------------------------------|
+| `[{<url>, <ref>, <use_case>}]` | A table of dependent applications to build for the `<test_use_case>` use case |
+
+This specify that in order to prepare for the `test_use_case`, the application located at `<url>` on ref `<ref>` should be compiled with its `<use_case>`.
```

Comment:
Inconsistent with the `Field name` header and could be misinterpreted as it is not clear `{<url>, <ref>, <use_case>}` refers to `key=value`.
I don't think we can do without explaining 1. each `tests.dependencies.*` is a list and 2. here are the fields of each element of this list.

~~~suggestion
The syntax to describe a single side application is the following:

| Field name | Description                                                           |
|------------|-----------------------------------------------------------------------|
| `url`      | The URL of the application git repository  |
| `ref`      | The reference to checked out in the repository |
| `use_case` | The `use_case` to use to build the application |

This is to be included in a list of applications attached to a particular testing scenario, for instance:
```toml
[[tests.dependencies.test_use_case]]
url = <url>
ref = <ref>
use_case = <use_case>
```
This specifies that the `test_use_case` scenario needs the application located at `<url>` on ref `<ref>` and should be built according to its `<use_case>`.

This following syntax can be used, as it is lighter and equivalent to the previous one:
```toml
[tests.dependencies]
test_use_case = {url = <url>, ref = <ref>, use_case = <use_case>}
```
~~~